### PR TITLE
Migrate admin area <Switch/> to RR6

### PR DESCRIPTION
**Pull request by philipp-spiess**
Part of #33834 

Here we go again.
 
## Test plan

- Tested locally
- Relying on CI

## App preview:

- [Web](https://sg-web-ps-migrate-admin-area-routes.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.04% (+1.23 kb) | 0.01% (+1.39 kb) | 0.00% (+0.16 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/migrate-admin-area-routes/compare-report.html;tab=live_object) for a full comparison between the commits cc70db4 and af83eee or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by valerybugakov** [comment]
@philipp-spiess, heads up, I fixed merged conflicts after merging my big PR into main.

File: client/web/src/site-admin/SiteAdminFeatureFlagConfigurationPage.tsx [hunk 1]
```diff
@@ -190,7 +191,7 @@ export const SiteAdminFeatureFlagConfigurationPage: FunctionComponent<
                                 name: flagName,
                             },
                         }).then(() => {
-                            history.push('../')
+                            navigate('../')
```
Comment (valerybugakov) [inline client/web/src/site-admin/SiteAdminFeatureFlagConfigurationPage.tsx:194]: I needed to use `navigate('..', { relative: 'path' })` to make it work as expected in `CodeIntelConfigurationPolicyPage.tsx`. Did you have a chance to test it?

File: client/web/src/site-admin/SiteAdminFeatureFlagConfigurationPage.tsx [hunk 2]
```diff
@@ -236,7 +237,7 @@ export const SiteAdminFeatureFlagConfigurationPage: FunctionComponent<
 
             <div className="mt-3">
                 {actions}
-                <Button type="button" className="ml-2" variant="secondary" onClick={() => history.push('../')}>
+                <Button type="button" className="ml-2" variant="secondary" onClick={() => navigate('../')}>
```
Comment (valerybugakov) [inline client/web/src/site-admin/SiteAdminFeatureFlagConfigurationPage.tsx:240]: Same as above

File: client/web/src/site-admin/SiteAdminWebhookCreatePage.story.tsx [hunk 3]
```diff
@@ -57,12 +56,7 @@ export const WebhookCreatePage: Story = () => {
         <WebStory>
             {() => (
                 <MockedTestProvider link={mocks}>
-                    <SiteAdminWebhookCreatePage
-                        match={{} as any}
-                        history={H.createMemoryHistory()}
-                        location={{} as any}
-                        telemetryService={NOOP_TELEMETRY_SERVICE}
-                    />
+                    <SiteAdminWebhookCreatePage telemetryService={NOOP_TELEMETRY_SERVICE} />
```
Comment (valerybugakov) [inline client/web/src/site-admin/SiteAdminWebhookCreatePage.story.tsx:59]: Nice

File: client/web/src/site-admin/outbound-webhooks/EditPage.story.tsx [hunk 4]
```diff
@@ -23,14 +22,9 @@ export const Page: Story = () => (
     <WebStory>
         {() => (
             <MockedTestProvider
-                link={new WildcardMockLink([logConnectionLink, buildOutboundWebhookMock('abcdef'), eventTypesMock])}
+                link={new WildcardMockLink([logConnectionLink, buildOutboundWebhookMock(''), eventTypesMock])}
```
Comment (valerybugakov) [inline client/web/src/site-admin/outbound-webhooks/EditPage.story.tsx:25]: We can also provide the initial URL to `WebStory` via `initialEntries`. Would it make sense to do it here to preserve the param match?

File: client/web/src/enterprise/executors/ExecutorsSiteAdminArea.tsx [hunk 5]
```diff
@@ -19,22 +19,22 @@ const GlobalExecutorSecretsListPage = lazyComponent<
...
-                path={`${match.url}/secrets`}
-                render={props => <GlobalExecutorSecretsListPage {...outerProps} {...props} />}
-                exact={true}
-            />
-            <Route render={() => <NotFoundPage pageType="settings" />} key="hardcoded-key" />
-        </Switch>
-    </>
-)
+export const ExecutorsSiteAdminArea: React.FunctionComponent<React.PropsWithChildren<Props>> = ({ ...outerProps }) => {
+    const url = '/site-admin/executors'
```
Comment (valerybugakov) [inline client/web/src/enterprise/executors/ExecutorsSiteAdminArea.tsx:26]:
```suggestion
const URL = '/site-admin/executors'

export const ExecutorsSiteAdminArea: React.FunctionComponent<React.PropsWithChildren<Props>> = outerProps => {
    
```

File: client/web/src/enterprise/site-admin/SiteAdminLsifUploadPage.tsx [hunk 6]
```diff
@@ -11,16 +11,13 @@ import { eventLogger } from '../../tracking/eventLogger'
...
 
 /**
  * A page displaying metadata about an LSIF upload.
  */
-export const SiteAdminLsifUploadPage: FunctionComponent<React.PropsWithChildren<Props>> = ({
-    match: {
-        params: { id },
-    },
-}) => {
+export const SiteAdminLsifUploadPage: FunctionComponent<React.PropsWithChildren<Props>> = () => {
```
Comment (valerybugakov) [inline client/web/src/enterprise/site-admin/SiteAdminLsifUploadPage.tsx:19]:
```suggestion
export const SiteAdminLsifUploadPage: FunctionComponent = () => {
```

File: client/web/src/enterprise/site-admin/SiteAdminLsifUploadPage.tsx [hunk 7]
```diff
@@ -11,16 +11,13 @@ import { eventLogger } from '../../tracking/eventLogger'
 
 import { fetchLsifUpload } from './backend'
 
-interface Props extends RouteComponentProps<{ id: string }> {}
+interface Props {}
```
Comment (valerybugakov) [inline client/web/src/enterprise/site-admin/SiteAdminLsifUploadPage.tsx:14]:
```suggestion
```

**Review by valerybugakov — APPROVED** [review]
Awesome work! We're close to the finish line 🤞

**Pull request merged**